app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1292,6 +1294,35 @@ export default Controller.extend(bufferedProperty("model"), {
     this.model.destroy(this.currentUser);
   },
 
+  retryOnRateLimit(times, promise, topicId) {
+    const currentTopicId = this.get("model.id");
+    topicId = topicId || currentTopicId;
+    if (topicId !== currentTopicId || times <= 0) {
+      // we navigated to another topic, so skip
+      return;
+    }
+
+    promise().catch((e) => {
+      const xhr = e.jqXHR;
+      if (
+        xhr &&
+        xhr.status === 429 &&
+        xhr.responseJSON &&
```

Comment:
I think you can use optional chaining here:

`xhr?.responseJson?.extras?.wait_seconds`

Let me know if it doesn't work

Reply:
Oh nice we have optional chaining

Reply:
> Oh nice we have optional chaining

yes be sure to read https://github.com/discourse/discourse/commit/bbddce4d3aa12bad4213ed69efa42dd01a844092 to know what's available now

I will merge a commit this week to make it available everywhere, but core should be good.